iothub/device/samples/convention-based-samples/TemperatureController/SystemTextJsonPayloadConvention.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Shared;
+
+namespace Microsoft.Azure.Devices.Client.Samples
+{
+    /// <summary>
+    /// A <see cref="PayloadConvention"/> that uses <see cref="SystemTextJsonPayloadSerializer"/>.
+    /// </summary>
+    public class SystemTextJsonPayloadConvention : PayloadConvention
```

Comment:
Why make this class?

Reply:
TemperatureControllerSample uses System.Text.Json as its object serializer while Thermostat uses NewtonSoft.Json. I thought we'd demonstrate both options this way. While we do provide a SystemTextJsonSerializer in our assembly, we won't be providing the corresponding Convention implementation. We'll only provide a NewtonSoft based convention in the assembly, by default.

Reply:
Why not provide a convention impl?

Reply:
No reason, we can add one if it'll be helpful. Since building the serializer is the most involved process, we added a sample for only that step.

Reply:
There are a bunch of customizations that are required for system.text.json, so we're planning to ship preview v1 without system.text.json in our assembly, but rather to add it to samples. Subsequently we'll promote system.text.json based serializer and convention to the assembly itself. What do you say, @drwill-ms ?